Woe, Super Synths be upon ye.

StarboundSolaris opened the pull request:
<!-- If you are new to the Delta-V repository, please read the [Contributing Guidelines](https://github.com/DeltaV-Station/Delta-v/blob/master/CONTRIBUTING.md) -->

## About the PR
<!-- What did you change? -->
Added Super Synths to the trinkets so that musician players with high amounts of playtime can get a treat for having to deal with waiting for epi to research Advanced Entertainment (and get a DAW built) for many, many shifts.
Super synths now can be worn on the neck so that you can use your hands for other things. (Security Officers can now have boss music as they go to kick the clowns ass for lube bombing med)
## Why / Balance
<!-- Discuss how this would affect game balance or explain why it was changed. Link any relevant discussions or issues. -->
Musicians should get a reward for playing the role for significant amounts of time, and this should also encourage people to develop their own midis for the server (or just encourage musicians to find good midis)
Also did this because I thought it was dumb that the best and most useful instrument in our codebase was locked behind staff perms and only really able to be spawned in by curators/admins. (It's literally just an instrument?!?!)
## Technical details
<!-- Summary of code changes for easier review. -->
Partially ports some starlight code to make this work, but...
Forced the super synth to be limited on how many notes they can play at once, so no more crashing the server with Rush E.
Added the ability for the super synth to be worn on the neck slot, because starlight has it and its cool.
Also locked the super synths behind 200 hours of musician playtime because I want them to suffer. :trollface: 
## Media
<!-- Attach media if the PR makes ingame changes (clothing, items, features, etc).
Small fixes/refactors are exempt. -->
<img width="729" height="706" alt="image" src="https://github.com/user-attachments/assets/ba40c8e3-fe05-484e-a4ff-682fff9a5e8b" />

## Requirements
<!-- Confirm the following by placing an X in the brackets: [X] -->
- [/] I have tested all added content and changes.
- [/] I have added media to this PR or it does not require an ingame showcase.
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->

## Breaking changes
<!-- List any breaking changes, including namespaces, public class/method/field changes, prototype renames; and provide instructions for fixing them. -->
Midis are currently broken due to upstream changes but that won't matter much.
**Changelog**
<!-- Add a Changelog entry to make players aware of new features or changes that could affect gameplay.
Make sure to read the guidelines and take this Changelog template out of the comment block in order for it to show up.
Changelogs must have a :cl: symbol, so the bot recognizes the changes and adds them to the game's changelog. -->
:cl:
- add: Added Super Synthesizers to trinkets! Musicians finally get a reward for all of their pain and suffering!
- add: Super Synthesizers can now be worn on the neck... Is it me or does that RPG-Wielding Captain have their own boss music?

StarboundSolaris committed:
Free Super synths for those who have suffered like I have.

StarboundSolaris committed:
Update instrument_keyed.yml

StarboundSolaris committed:
Update meta.json

StarboundSolaris committed:
Update meta.json

StarboundSolaris commented:
Forgot to mention, but this PR should allow for mappers/event runners to use the supersynth more freely since it now has limits to what it can play.

Cepelinas1 commented:
direction review 24h

StarboundSolaris commented:
> direction review 24h

<img width="482" height="349" alt="LIVE IZUNIA REACTION" src="https://github.com/user-attachments/assets/ee9fcee4-492a-4428-9e79-de9ba009a643" />

jimmy12or commented:
Does this unlock it for any role as long as the person has had a lot of musician time, or is this job-specific? If it's not job-specific, i feel it should be so that not everyone is going around blaring their own music on a super-synth and making it harder for the musician to play what they would like.

StarboundSolaris commented:
> Does this unlock it for any role as long as the person has had a lot of musician time, or is this job-specific? If it's not job-specific, i feel it should be so that not everyone is going around blaring their own music on a super-synth and making it harder for the musician to play what they would like.

It unlocks it as a trinket for use with all jobs. The high playtime requirement is meant for it to be seen not as often in rounds, as well as to ensure that it isn't just being given out like Halloween candy to every random musician on the server. And I will also mention that people already go around blaring their own music anyways since instruments are _incredibly_ common to obtain, and that has always been an issue both in delta-V and outside the server (hell, even with instruments in SS13 as well.). I will also mention that at least here, people are somewhat respectful and at least _try_ to not play over the musicians music, or at least just move away so that they aren't interrupting them.

StarboundSolaris commented:
With the playtime requirement, it would require you to play at least 50-60 rounds (assuming they all last for close to 2 hours) worth of musician shifts. If it becomes an issue to the point that everyone somehow has a supersynth (or are farming musician time for it, which can generally just be handled by administration, since playtime farming is not allowed at all), ill just increase the time requirements further.

Cepelinas1 commented:
fell asleep, oops
direction approved for a test merge, incase these synths crash the server

StarboundSolaris commented:
> fell asleep, oops direction approved for a test merge, incase these synths crash the server

dont worry, they shouldnt crash unless someone is admin abusing and uses VV to re-disable their limits.

StarboundSolaris commented:
@deltanedas Heya, could you review this real quick!

deltanedas reviewed (dismissed):
comment things properly you didnt comment all changes to upstream stuff

and leave meta.json indent alone its conflict bait

StarboundSolaris committed:
Merge branch 'DeltaV-Station:master' into MusicianSuperSynthChaos

StarboundSolaris commented:
> comment things properly you didnt comment all changes to upstream stuff
> 
> and leave meta.json indent alone its conflict bait

Meta.json indent should be fine unless jetbrains fucked up the indent again.

StarboundSolaris commented:
Either way ill look into it later.

StarboundSolaris commented:
@deltanedas may just be a github mobile app issue, but can you elaborate on what upstream stuff i  forgot to comment?

StarboundSolaris commented:
As in i cant see on the app if you pointed out where already

StarboundSolaris commented:
Just says requested changes only

StarboundSolaris commented:
okay yeah i pulled the pr up on my laptop and I can see it now, and it looks like jetbrains fucked with the indents again...

StarboundSolaris commented:
also @deltanedas you may have to walk me through on what upstream stuff you said needed comments, since im having a brain fart and cannot figure out what you want commented.

StarboundSolaris commented:
sorry misclick, ignore that

deltanedas commented:
size normal

StarboundSolaris committed:
Doubled Playtime because I realized that 100 hours is not long enough considering that musicians are almost always being played on delta.

Doubled playtime for obvious reasons, also because 100 hours was not evil enough.

StarboundSolaris commented:
> size normal

uhhhh one moment

StarboundSolaris committed:
Comment moment.

StarboundSolaris commented:
@deltanedas jetbrains is having a stroke and not fixing my indent do you remember what the thing to fix it was again.

StarboundSolaris committed:
Merge branch 'master' into MusicianSuperSynthChaos

StarboundSolaris commented:
Also dont ask why the size normal thing is there, it just needs it to render on screen properly and I am literally too scared to investigate why, since it just doesnt work without it. Presuming thats an upstream issue or something.

StarboundSolaris committed:
I think this is the fix, please yell at me if it isnt deltanedas.

deltanedas commented:
comment changes still

and im not your fucking tutor figure out how to use a text editor

StarboundSolaris commented:
> comment changes still
> 
> and im not your fucking tutor figure out how to use a text editor

Jeez, alright.

StarboundSolaris committed:
comments...

StarboundSolaris committed:
comment stuff 2

StarboundSolaris committed:
messing with stuff, one sec.

StarboundSolaris committed:
testing...

StarboundSolaris committed:
i hate it here, may have finally fixed the bullshit with indents

StarboundSolaris committed:
Merge branch 'master' into MusicianSuperSynthChaos

StarboundSolaris commented:
> With the playtime requirement, it would require you to play at least 50-60 rounds (assuming they all last for close to 2 hours) worth of musician shifts. If it becomes an issue to the point that everyone somehow has a supersynth (or are farming musician time for it, which can generally just be handled by administration, since playtime farming is not allowed at all), ill just increase the time requirements further.

Update on this, I went in and doubled the playtime requirement to 200 hours anyways, so make that 100-120 rounds worth of musician shifts. Should ensure that super synths are used only by those worthy to use them. (and hopefully makes them somewhat rare to see in rounds.) This should automatically unlock for those who already have the 200 hours, so those folks shouldnt need to play for another 200 hours to unlock this..

StarboundSolaris committed:
Merge branch 'master' into MusicianSuperSynthChaos

StarboundSolaris commented:
Requesting re review.

StarboundSolaris committed:
Merge branch 'master' into MusicianSuperSynthChaos

StarboundSolaris committed:
Merge branch 'master' into MusicianSuperSynthChaos

StarboundSolaris commented on Resources/Textures/Objects/Fun/Instruments/h_synthesizer.rsi/meta.json:
Unless theres some way of putting comments on JSON files without it causing the game to freak out I dont know if this is fine or not.

StarboundSolaris replied:
should be fine. probably. is just a copypasted sprite of one of the inhand sprites.

TehFlaminTaco commented on Resources/Prototypes/Entities/Objects/Fun/Instruments/instruments_percussion.yml:
```diff
@@ -54,7 +54,7 @@
       "Aah": {52: 0}
       "Ooh": {53: 0}
-      # "Kweh": {4: 1} #Delta-V change, this stupid instrument fucks with literally every single midi known to mankind. Uncomment when whoever added this makes this not fuck with every midi. - Solaris
+      # "Kweh": {4: 1} # Delta-V change - this stupid instrument fucks with literally every single midi known to mankind. Uncomment when whoever added this makes this not fuck with every midi. - Solaris
```
Untouch, shouldn’t edit this file

StarboundSolaris replied:
fixed.

TehFlaminTaco commented on Resources/Prototypes/Loadouts/loadout_groups.yml:
```diff
@@ -67,6 +67,9 @@
   - Eyepatch
   # End DeltaV Additions
+  # Begin Starlight Additions
+  - SuperSynthesizerInstrument # 🌟Starlight🌟
```
Move the starlight comment to the end of the line, as there’s only one

StarboundSolaris replied:
Think I did this right, unsure what you meant by the end of the line, so I just moved it to the starlight additions stuff.

TehFlaminTaco replied:
It doesn't need the `# Begin Starlight Additions` and End comments

StarboundSolaris replied:
Ahh okay, will go in and fix that real quick.

StarboundSolaris replied:
fixed.

TehFlaminTaco commented on Resources/Textures/Objects/Fun/Instruments/h_synthesizer.rsi/meta.json:
```diff
@@ -28,6 +28,10 @@
     {
       "name": "super-inhand-right",
       "directions": 4
+    },
```
Missing attribution, unless this has the exact same source as the others.

StarboundSolaris replied:
has same source as others, should be fine.

StarboundSolaris committed:
requested changes

StarboundSolaris committed:
Merge branch 'master' into MusicianSuperSynthChaos

StarboundSolaris committed:
Merge branch 'master' into MusicianSuperSynthChaos

StarboundSolaris committed:
Requested Changes

TehFlaminTaco reviewed (approved):
lgtm

StarboundSolaris commented:
it begins.
